src/utils/ao.ts

```diff
@@ -157,7 +158,43 @@ export async function evolveANT({
   return id;
 }
 
+export function isAoSigner(value: unknown): value is AoSigner {
+  const TagSchema = z.object({
+    name: z.string(),
+    value: z.union([z.string(), z.number()]),
+  });
```

Comment:
does aoconnect expose any of these schemas?

Reply:
It has them but doesn't expose them - and even if it did, they are inaccurate. They claim tag values can only be strings, and they use `any` for the data type.

Reply:
FWIW tags must be converted to strings when posting to arweave, so I wouldn't go sofar as to say their types are inaccurate